schematic/schemas/json_schema_validation_rule_functions.py

```diff
@@ -0,0 +1,367 @@
+            ValidationRuleName.URL,
+            ValidationRuleName.DATE,
+            ValidationRuleName.REGEX,
+            ValidationRuleName.STR,
+            ValidationRuleName.BOOL,
+        ],
+        parameters=["minimum", "maximum"],
+    ),
+    "str": ValidationRule(
+        name=ValidationRuleName.STR,
+        js_type=JSONSchemaType.STRING,
+        incompatible_rules=[
+            ValidationRuleName.IN_RANGE,
+            ValidationRuleName.INT,
+            ValidationRuleName.FLOAT,
+            ValidationRuleName.NUM,
+            ValidationRuleName.BOOL,
+        ],
+    ),
+    "float": ValidationRule(
+        name=ValidationRuleName.FLOAT,
+        js_type=JSONSchemaType.NUMBER,
+        incompatible_rules=[
+            ValidationRuleName.URL,
+            ValidationRuleName.DATE,
+            ValidationRuleName.REGEX,
+            ValidationRuleName.STR,
+            ValidationRuleName.BOOL,
+        ],
+    ),
+    "int": ValidationRule(
+        name=ValidationRuleName.INT,
+        js_type=JSONSchemaType.INTEGER,
+        incompatible_rules=[
+            ValidationRuleName.URL,
+            ValidationRuleName.DATE,
+            ValidationRuleName.REGEX,
+            ValidationRuleName.STR,
+            ValidationRuleName.BOOL,
+        ],
+    ),
+    "num": ValidationRule(
+        name=ValidationRuleName.NUM,
+        js_type=JSONSchemaType.NUMBER,
+        incompatible_rules=[
+            ValidationRuleName.URL,
+            ValidationRuleName.DATE,
+            ValidationRuleName.REGEX,
+            ValidationRuleName.STR,
+            ValidationRuleName.BOOL,
+        ],
+    ),
+}
+
+
+def filter_unused_inputted_rules(inputted_rules: list[str]) -> list[str]:
+    """Filters a list of validation rules for only those used to create JSON Schemas
+
+    Arguments:
+        inputted_rules: A list of validation rules
+
+    Returns:
+        A filtered list of validation rules
+    """
+    return [
+        rule
+        for rule in inputted_rules
+        if _get_name_from_inputted_rule(rule) in [e.value for e in ValidationRuleName]
+    ]
+
+
+def check_for_duplicate_inputted_rules(inputted_rules: list[str]) -> None:
+    """Checks that there are no duplicate rules by name
+
+    Arguments:
+        inputted_rules: A list of validation rules
+
+    Raises:
+        ValueError: If there are multiple rules with the same name
+    """
+    rule_names = _get_names_from_inputted_rules(inputted_rules)
+    if sorted(rule_names) != sorted(list(set(rule_names))):
+        raise ValueError("Validation Rules contains duplicates: ", inputted_rules)
+
+
+def check_for_conflicting_inputted_rules(inputted_rules: list[str]) -> None:
+    """Checks that each rule has no conflicts with any other rule
+
+    Arguments:
+        inputted_rules: A list of validation rules
+
+    Raises:
+        ValueError: If a rule is in conflict with any other rule
+    """
+    rule_names = _get_names_from_inputted_rules(inputted_rules)
```

Comment:
Since `inputted_rules` are the same for both `check_for_conflicting_inputted_rules` and `check_for_duplicate_inputted_rules`, I am wondering can this step be called only once?